src/include/miopen/batchnorm/problem_description.hpp

```diff
@@ -222,10 +222,47 @@ struct MIOPEN_INTERNALS_EXPORT ProblemDescription : ProblemDescriptionBase, Prob
     std::string in_layout  = "NCHW";
     std::string out_layout = "NCHW";
     std::string din_layout = "NCHW";
+    int spatial_dim        = 2;
```

Comment:
Looks like the number of dimensions is now hardcoded. How will  3d cases be handled?